numpy/linalg/linalg.py

```diff
@@ -2110,7 +2110,7 @@ def lstsq(a, b, rcond="warn"):
     if is_1d:
         b = b[:, newaxis]
     _assertRank2(a, b)
-    _assertNoEmpty2d(a, b)  # TODO: relax this constraint
+    is_empty_b = _isEmpty2d(b)
```

Comment:
Should this test really be `n_rhs == 0`?

Reply:
This requires some thought in the general case... if `b` were 1D, it would have to have the same number of entries as `a`, so the check for compatible dimensions would make things work. As for the rest, it depends on how the prospective broadcasting is to be done.

Reply:
I suppose my point is that adding `1` to `n_rhs` is a waste of time if `n_rhs` was already not zero.

Reply:
Well, even if it were zero, we still need to do the `svd` to get the rank and singular values of `a`. And we won't increase `n_rhs` if it is non-zero.

Reply:
> And we won't increase n_rhs if it is non-zero.

Sure you will - when `b.shape = (0, 3)` (ie m == 0, n_rhs == 3), you resize b to (0, 4) even though there's no reason to. The fact that you agree with my comment "lapack can't handle n_rhs = 0 - so allocate the array one larger" suggests that the if condition should be simply `n_rhs == 0`

Reply:
got it